sql/core.go

```diff
@@ -288,6 +288,14 @@ type OpaqueNode interface {
 	Opaque() bool
 }
 
+// Projector is a node that projects expressions for parent nodes to consume (i.e. GroupBy, Window, Project).
+type Projector interface {
+	// ProjectedExprs returns the list of expressions projected by this node.
+	ProjectedExprs() []Expression
+	// WithProjectedExprs returns a new Projector instance with the specified expressions set as its projected expressions.
```

Comment:
nit: the docstring says we return a Projector, but the interface is Node

Reply:
good catch; fixed to return `sql.Projector`